### ISSUE TITLE
Lag when copy/pasting from Illustrator

### ISSUE THREAD

**Issue by DCNielsen** [folivoraAI/BetterTouchTool#2002]
Mac 2015-Retina. OS X 10.12.6. BTT 2.315

Every time I copy or paste in Illustrator there's a monster lag. If I quit BTT, the problem disappears. This makes me sad because BTT is a huge part of my Illustrator workflow. 

It's happened before, but the issue was on Adobe's end because they broke Illustrator for every Clipboard Management app. They pushed a fix with a recent update and I haven't had problems. Without updating Illustrator or anything else (I thought), the problem started up again this week. Whenever I copy or paste, there's a 5 second Pinwheel of death before I get control of the interface again.

Could a light-weight bandaid solution be a choice to turn off Clipboard Management for specific apps?

Thank you Andreas!

**Comment by fifafu** [comment]
Hey,

I'll look into this, but probably won't be able to look into the details until early February. For now disabling the BetterTouchTool clipboard manager for Illustrator should fix this. (Just click the BetterTouchTool menubar icon and select "disable clipboard manager for current application)

Best
Andreas

**Comment by DCNielsen** [comment]
Thanks. I just saw that this feature exists in the menubar a minute before I saw your post! It's a good fix for now. Wish I could still use the Clipboard Manager but at least I don't have to choose between BTT or lag. When you get to debugging this, Let me know if you want me to see if other Clipboard Managers are giving me the same hiccup, etc.

**Comment by kiva113** [comment]
I have been noticing the lag showing up in Excel when copying and pasting rows. Often I need to force quit it. 

It seems to be a problem if you have multiple programs that have the clipboard history added feature running. I have confirm this by closing out other programs and the issue goes away. Unfortunately not all programs allow you to disable it, or never actually stop tracking in the background.

Can you add a feature to disable the tracking completely?

BTW, I'm running Sierra patched with the Spectre/Meltdown fix. I would need someone else to verify, but this issue may have started for me after the update.

**Comment by fifafu** [comment]
What exactly do you mean by disable the tracking completely? If you want to disable the clipboard manager you just need to remove any trigger you have set up to show it. (Maybe restart BTT afterwards).

I'll look into this issue in the next few days :-)

**Comment by kiva113** [comment]
I just dug in and realized I still had one gesture set and once I removed that it does appear to have completely disabled the clipboard manager; ie. the option to disable the clipboard manager for a specific application in the drop-down menu was gone.

**Comment by fifafu** [comment]
I just stumbled upon this thread: https://forums.adobe.com/thread/2171569
It seems like the clipboard issues can be solved by disabling "include svg code" in the Illustrator options fixed a similar issue for users of other clipboard managers.

![image](https://user-images.githubusercontent.com/8820811/37252146-a531b50c-251c-11e8-90ab-c4798eb1ecb4.png)

Also apparently Adobe has fixed it in an update for Illustrator - which version are you currently running?

**Comment by DCNielsen** [comment]
Hi! I believe when I was having this problem I was using the most recent version of Illustrator (at the time - late January). I had already experienced the problem previously and it went away with the update before that - then came back with the update closest to my bug-posting. 

To test, I've re-enabled BTT and copy-pasted in Illustrator v 22.0.1. The bug's not reproducing. No problems today. Also, if I enable Include SVG Code, I'm not having problems either.

I seem to recall that when I first experienced this bug, unchecking "Include SVG Code" fixed it.

Sorry for the spotty data, but it's working as of today. I'll keep BTT and SVG Code on and we'll see if I hit another glitch.
